sdk/identity/Azure.Identity/src/ScopeUtilities.cs

```diff
@@ -13,7 +13,7 @@ namespace Azure.Identity
     internal static class ScopeUtilities
     {
         private const string DefaultSuffix = "/.default";
-        private const string ScopePattern = "^[0-9a-zA-Z-.:/]+$";
+        private const string ScopePattern = "^[0-9a-zA-Z-_.:/]+$";
```

Comment:
Should this validation be used for all sub-process based credentials? It looks like only Cli and Powershell use it today.

Reply:
Yes, it should, although it might not be at the moment. Also, when I made this update, I did a brief investigation I found that there are other possible scope characters that this regex is still not supporting. We need to do a deeper investigation of both, but given that we have 2 customers blocked by this, and `user_impersonation` is such a common scope, I think we should push out this minor fix although it might be incomplete. I created a new issue https://github.com/Azure/azure-sdk-for-net/issues/31170, to track what work remains here.